parts/k8s/cloud-init/artifacts/cse_helpers.sh

```diff
@@ -227,5 +227,11 @@ sysctl_reload() {
 version_gte() {
   test "$(printf '%s\n' "$@" | sort -rV | head -n 1)" == "$1"
 }
-
+exit_cse() {
+  local output=${@: -2: 1}
+  local exit_code=${@: -1: 1}
+  local command=${@:1:$#-2}
+  $command >> $output
```

Comment:
You might want to make sure that you don't care about the exit/failure in line 234 (the $command part)
Including that the $command is slow.  I would think maybe backgrounding that such that it can complete separate from the exit (just a thought)

Something like:  `$command >> $output &`

The point being that you would like this information but you don't want it to cause a nested failure.

Reply:
I like the idea of running in the background